RadixWallet/Features/AccountDetailsFeature/HeaderListViewContainer.swift

```diff
@@ -0,0 +1,119 @@
+import SwiftUI
+@_spi(Advanced) import SwiftUIIntrospect
+
+/**
+ A component that creates a collapsible header with a list.
+
+ This `HeaderListViewContainer` struct is a `UIViewRepresentable` wrapper for combining a header view and a list view.
+
+ - Parameters:
+    - Header: A `View` representing the header.
+    - List: A `View` representing the list.
+
+ This component dynamically adjusts the layout based on the scroll position of the list. As the user scrolls, the header collapses and fades out, providing a smooth transition effect.
+
+ - Note: This component relies on the Introspect library for UIKit introspection.
+ */
+public struct HeaderListViewContainer<Header: View, List: View>: UIViewRepresentable {
+	public typealias UIViewType = UIView
+
+	private let headerView: Header
+	private let listView: List
+
+	@State private var observation: NSKeyValueObservation?
+
+	public init(
+		@ViewBuilder headerView: () -> Header,
+		@ViewBuilder listView: () -> List
+	) {
+		self.headerView = headerView()
+		self.listView = listView()
+	}
+
+	public func makeUIView(context: Context) -> UIView {
+		let containerView = UIView()
+		addSubviews(in: containerView)
+
+		return containerView
+	}
+
+	public func updateUIView(_ uiView: UIViewType, context: Context) {
+		uiView.subviews.forEach { $0.removeFromSuperview() }
+		addSubviews(in: uiView)
```

Comment:
do we need to remove and add every subview each time? It is adding a small glitch every time the view appears/disappears (it shows the activity indicator for an instant).


https://github.com/radixdlt/babylon-wallet-ios/assets/164921079/8649d9b9-4f3d-40eb-9a8f-2c15af5aff69


![Screenshot 2024-04-02 at 16 27 46](https://github.com/radixdlt/babylon-wallet-ios/assets/164921079/5da45bf9-2a6c-4971-b756-8c8cbb6f1869)